src/odemis/driver/actuator.py

```diff
@@ -1479,9 +1481,13 @@ def _readPositionfromChildren(self):
         """
         pos_children_cur = [c.position.value[ca] for c, ca in zip(self._children, self._caxes_map)]
 
-        # check whether children positions are in consistency with positions allowed
+        # check whether current children positions are in consistency with positions allowed
         for pos_key, pos_children in self._positions.items():
-            for ap, cp, atol in zip(pos_children_cur, pos_children, self._atol):
+            for ap, cp, atol, cyl in zip(pos_children_cur, pos_children, self._atol, self._cycle):
+                # handle positions close to cycle and thus also close to zero
+                # and within tolerance so util.almost_equal compare the correct values
+                if cyl - ap < atol:
+                    ap = cyl - ap
```

Comment:
I don't think it handles correctly the reverse case where cp is almost as large as cyl (eg, 6.2) and ap is > 0.
I think that you could write (but you need to check):
```
if cyl is None:
    dist = abs(ap - cp)
else:
    dist = min((ap - cp) % cyl, (cp - ap) % cyl)

if dist > atol:
    break
```